cmake/scripts/darwin_embedded/ArchSetup.cmake

```diff
@@ -1,34 +1,37 @@
 if(NOT CMAKE_TOOLCHAIN_FILE)
-  message(FATAL_ERROR "CMAKE_TOOLCHAIN_FILE required for ios. See ${CMAKE_SOURCE_DIR}/cmake/README.md")
+  message(FATAL_ERROR "CMAKE_TOOLCHAIN_FILE required for ios/tvos. See ${CMAKE_SOURCE_DIR}/cmake/README.md")
 endif()
 
-set(CORE_MAIN_SOURCE ${CMAKE_SOURCE_DIR}/xbmc/platform/darwin/${CORE_PLATFORM_NAME}/XBMCApplication.mm)
+set(CORE_MAIN_SOURCE ${CMAKE_SOURCE_DIR}/xbmc/platform/darwin/${CORE_PLATFORM_NAME_LC}/XBMCApplication.mm)
+set(PLATFORM_BUNDLE_INFO_PLIST ${CMAKE_SOURCE_DIR}/xbmc/platform/darwin/${CORE_PLATFORM_NAME_LC}/Info.plist.in)
 
-set(ARCH_DEFINES -DTARGET_POSIX -DTARGET_DARWIN -DTARGET_DARWIN_EMBEDDED)
+if(CORE_PLATFORM_NAME_LC STREQUAL tvos)
+  set(ARCH_DEFINES -DTARGET_DARWIN_TVOS)
+  set(CMAKE_XCODE_ATTRIBUTE_TARGETED_DEVICE_FAMILY "3")
+else()
+  set(ARCH_DEFINES -DTARGET_DARWIN_IOS)
+  set(CMAKE_XCODE_ATTRIBUTE_TARGETED_DEVICE_FAMILY "1,2")
+endif()
+list(APPEND ARCH_DEFINES -DTARGET_POSIX -DTARGET_DARWIN -DTARGET_DARWIN_EMBEDDED)
```

Comment:
set the common ARCH_DEFINES first, and the add the platform specific ones at the end (move this line above the if block)